go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/argoproj-labs/terraform-provider-argocd
 
-go 1.22.7
+go 1.23.0
```

Comment:
We should update to go 1.23 in a combined PR and not within the PR of a single dependency.